### PR TITLE
[IE][VPU] Support S32 input for StridedSlice

### PR DESCRIPTION
**Pull request by adudchen**
#-33421

### CODE REVIEW

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/strided_slice.cpp [hunk 1]
```diff
@@ -11,90 +11,64 @@ using namespace LayerTestsDefinitions;
...
-                                InferenceEngine::Precision::I64, CommonTestUtils::DEVICE_GPU),
-        stridedSliceParamsTuple({ 2, 2, 2, 2 }, { 1, 1, 1, 1 }, { 2, 2, 2, 2 }, { 1, 1, 1, 1 },
-                                {0, 0, 0, 0}, {0, 0, 0, 0},  {},  {},  {},
-                                InferenceEngine::Precision::I64, CommonTestUtils::DEVICE_GPU),
-        stridedSliceParamsTuple({ 2, 3, 4, 5, 6 }, { 0, 1, 0, 0, 0 }, { 2, 3, 4, 5, 6 }, { 1, 1, 1, 1, 1 },
-                                {1, 0, 1, 1, 1}, {1, 0, 1, 1, 1},  {},  {0, 1, 0, 0, 0},  {},
-                                InferenceEngine::Precision::FP32, CommonTestUtils::DEVICE_GPU),
+InferenceEngine::Precision precision[] = {InferenceEngine::Precision::FP32, InferenceEngine::Precision::I64};
+
+stridedSliceSpecificParams ss_only_test_cases[] = {
```
Comment (gladilov-gleb) [inline inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/strided_slice.cpp:16]: Did you remove some test cases?

File: inference-engine/src/vpu/graph_transformer/src/stages/strided_slice.cpp [hunk 2]
```diff
@@ -39,14 +39,16 @@ class StridedSliceStage final : public StageNode {
...
-            { {DataType::FP16}, {DataType::S32}, {DataType::S32}, {DataType::S32} };
+            { {DataType::FP16, DataType::S32}, {DataType::S32}, {DataType::S32}, {DataType::S32} };
+
+        const auto outType = input(0)->desc().type();
 
         assertInputsOutputsTypes(
             this,
             numInputs() == 3 ? expectedInputs3Types : expectedInputs4Types,
-            {{DataType::FP16}});
+            {{outType}});
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/stages/strided_slice.cpp:51]: Do we expect that first input precision and output precision are the same without bounding to FP16 or S32?

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/strided_slice.cpp (hunk 1 above)
Comment (adudchen) [reply inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/strided_slice.cpp:16]: No, these are the same. I removed three identical tests which differed only in precision.

**Comment by Maxim-Doronin** [comment]
Closed due refactoring StridedSlice and adding StridedSlice kernel on the device side which supports s32 inputs
https://github.com/openvinotoolkit/openvino/pull/1244

**Pull request closed without merging**